<!DOCTYPE html> <html lang="en-US"> <head> <meta charset="UTF-8"> <meta http-equiv="X-UA-Compatible" content="IE=Edge"> <link rel="stylesheet" href="/YAME/assets/css/just-the-docs-default.css"> <link rel="stylesheet" href="/YAME/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"> <style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li:not(:nth-child(1)) > a, .site-nav > ul.nav-list:first-child > li > ul > li a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(1) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(1) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(1) > ul.nav-list { display: block; } </style> <script src="/YAME/assets/js/vendor/lunr.min.js"></script> <script src="/YAME/assets/js/just-the-docs.js"></script> <meta name="viewport" content="width=device-width, initial-scale=1"> <!-- Begin Jekyll SEO tag v2.8.0 --> <title>Home | YAME</title> <meta name="generator" content="Jekyll v4.3.4" /> <meta property="og:title" content="Home" /> <meta property="og:locale" content="en_US" /> <meta name="description" content="YAME (Yet Another Methylation Encoder) is a fast and lightweight toolkit for storing, manipulating, and analyzing large-scale DNA methylation data at the sequence level. It introduces the CX format family for ultra-compact representation of methylation values, states, and genomic coordinates." /> <meta property="og:description" content="YAME (Yet Another Methylation Encoder) is a fast and lightweight toolkit for storing, manipulating, and analyzing large-scale DNA methylation data at the sequence level. It introduces the CX format family for ultra-compact representation of methylation values, states, and genomic coordinates." /> <link rel="canonical" href="http://localhost:4000/YAME/" /> <meta property="og:url" content="http://localhost:4000/YAME/" /> <meta property="og:site_name" content="YAME" /> <meta property="og:type" content="website" /> <meta name="twitter:card" content="summary" /> <meta property="twitter:title" content="Home" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebSite","description":"YAME (Yet Another Methylation Encoder) is a fast and lightweight toolkit for storing, manipulating, and analyzing large-scale DNA methylation data at the sequence level. It introduces the CX format family for ultra-compact representation of methylation values, states, and genomic coordinates.","headline":"Home","name":"YAME","url":"http://localhost:4000/YAME/"}</script> <!-- End Jekyll SEO tag --> </head> <body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"> <title>Link</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"> <path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"> <title>Menu</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"> <line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"> <title>Expand</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"> <polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <!-- Feather. MIT License: https://github.com/feathericons/feather/blob/master/LICENSE --> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"> <title id="svg-external-link-title">(external link)</title> <path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"> <title>Document</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"> <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"> <title>Search</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <!-- Bootstrap Icons. MIT License: https://github.com/twbs/icons/blob/main/LICENSE.md --> <symbol id="svg-copy" viewBox="0 0 16 16"> <title>Copy</title> <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"> <path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/> <path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"> <title>Copied</title> <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"> <path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/> <path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg> <div class="side-bar"> <div class="site-header" role="banner"> <a href="/YAME/" class="site-title lh-tight"> YAME </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button> </div> <nav aria-label="Main" id="site-nav" class="site-nav"> <ul class="nav-list"><li class="nav-list-item"><a href="/YAME/" class="nav-list-link">Home</a></li><li class="nav-list-item"><a href="/YAME/docs/summarize.html" class="nav-list-link">Summary and Encoding</a></li><li class="nav-list-item"><a href="/YAME/docs/storage.html" class="nav-list-link">Storage & Format</a></li><li class="nav-list-item"><a href="/YAME/docs/enrichment.html" class="nav-list-link">Enrichment Testing</a></li><li class="nav-list-item"><a href="/YAME/docs/pack_unpack.html" class="nav-list-link">Storage & Format</a></li><li class="nav-list-item"><a href="/YAME/docs/subset.html" class="nav-list-link">Methylation Data Subset</a></li><li class="nav-list-item"><a href="/YAME/docs/rowop.html" class="nav-list-link">Row Subset/Operations</a></li><li class="nav-list-item"><a href="/YAME/docs/sample.html" class="nav-list-link">Sample Combination/Indexing</a></li><li class="nav-list-item"><a href="/YAME/docs/dsample.html" class="nav-list-link">Downsampling</a></li><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in YAME Subcommands category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/YAME/docs/subcommands" class="nav-list-link">YAME Subcommands</a><ul class="nav-list"><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_pack.html" class="nav-list-link">yame pack</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_unpack.html" class="nav-list-link">yame unpack</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_subset.html" class="nav-list-link">yame subset</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_rowsub.html" class="nav-list-link">yame rowsub</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_info.html" class="nav-list-link">yame info</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_summary.html" class="nav-list-link">yame summary</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_index.html" class="nav-list-link">yame index</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_split.html" class="nav-list-link">yame split</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_chunk.html" class="nav-list-link">yame chunk</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_chunkchar.html" class="nav-list-link">yame chunkchar</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_rowop.html" class="nav-list-link">yame rowop</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_mask.html" class="nav-list-link">yame mask</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_dsample.html" class="nav-list-link">yame dsample</a></li><li class="nav-list-item"><a href="/YAME/docs/subcommands/YAME_pairwise.html" class="nav-list-link">yame pairwise</a></li></ul></li></ul> </nav> <footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll. </footer> </div> <div class="main" id="top"> <div id="main-header" class="main-header"> <div class="search" role="search"> <div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search YAME" aria-label="Search YAME" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label> </div> <div id="search-results" class="search-results"></div> </div> <nav aria-label="Auxiliary" class="aux-nav"> <ul class="aux-nav-list"> <li class="aux-nav-list-item"> <a href="//github.com/zhou-lab/YAME" class="site-button" > YAME on GitHub </a> </li> </ul> </nav> </div> <div class="main-content-wrap"> <div id="main-content" class="main-content"> <main> <h1 class="fs-9" id="yame---yet-another-methylation-encoder"> <a href="#yame---yet-another-methylation-encoder" class="anchor-heading" aria-labelledby="yame---yet-another-methylation-encoder"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> YAME - Yet Another Methylation Encoder </h1> <p class="fs-6 fw-300">A fast and lightweight toolkit for sequence-level DNA methylation analysis</p> <p><a href="#quick-start" class="btn btn-primary fs-5 mb-4 mb-md-0 mr-2">Get Started</a> <a href="https://github.com/zhou-lab/YAME" class="btn fs-5 mb-4 mb-md-0">View on GitHub</a></p><hr /> <h2 id="overview"> <a href="#overview" class="anchor-heading" aria-labelledby="overview"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Overview </h2> <p>YAME (Yet Another Methylation Encoder) is designed for efficient sequence-level DNA methylation data management, capable of handling both bulk and single-cell DNA methylome workflows. It introduces a family of compact binary formats (<strong>CX formats</strong>) that represent methylation values, MU counts, categorical states, fraction data, masks, and genomic coordinates in a uniform compressed structure.</p> <h3 id="key-features"> <a href="#key-features" class="anchor-heading" aria-labelledby="key-features"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Key Features </h3> <ul> <li><strong>ðŸš€ Ultra-fast performance</strong> - Efficient C implementation with BGZF compression</li> <li><strong>ðŸ’¾ High compression ratios</strong> - Compact binary CX formats dramatically reduce storage requirements</li> <li><strong>ðŸ“ˆ Massive scalability</strong> - Handle hundreds of thousands of single cells</li> <li><strong>ðŸ”§ Comprehensive toolkit</strong> - Pack, unpack, subset, downsample, merge, and analyze methylation data</li> <li><strong>ðŸ§¬ Versatile data support</strong> - MU counts, binary methylation, chromatin states, fractions, differential calls, and CpG coordinates</li> <li><strong>ðŸ”— Seamless integration</strong> - Works with bedtools, KYCG knowledge base, and standard methylation workflows</li> </ul> <h3 id="use-cases"> <a href="#use-cases" class="anchor-heading" aria-labelledby="use-cases"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Use Cases </h3> <ul> <li><strong>Single-cell methylome analysis</strong> - Efficiently store and analyze sparse single-cell data</li> <li><strong>Bulk methylome processing</strong> - Fast operations on large cohorts</li> <li><strong>Enrichment testing</strong> - Test methylation enrichment across genomic features</li> <li><strong>Feature aggregation</strong> - Summarize methylation over bins, chromatin states, or custom regions</li> <li><strong>Pseudobulk generation</strong> - Merge single cells into cluster-level pseudobulks</li> <li><strong>Differential methylation</strong> - Identify and test differentially methylated sites</li> </ul><hr /> <h2 id="quick-start"> <a href="#quick-start" class="anchor-heading" aria-labelledby="quick-start"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Quick Start </h2> <h3 id="installation"> <a href="#installation" class="anchor-heading" aria-labelledby="installation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation </h3> <p><strong>Option 1: Install via Conda (Recommended)</strong></p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>conda <span class="nb">install </span>yame <span class="nt">-c</span> bioconda
</code></pre></div></div> <p><strong>Option 2: Build from Source</strong></p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>git clone https://github.com/zhou-lab/YAME.git
<span class="nb">cd </span>YAME
make
</code></pre></div></div> <p>To run YAME from anywhere, move it to your PATH:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">mv</span> ./yame ~/bin/yame
</code></pre></div></div> <h3 id="basic-usage-example"> <a href="#basic-usage-example" class="anchor-heading" aria-labelledby="basic-usage-example"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Basic Usage Example </h3> <p><strong>1. Pack methylation data into CX format</strong></p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Pack binary methylation data</span>
yame pack <span class="nt">-fb</span> yourfile.bed <span class="o">&gt;</span> yourfile.cg

<span class="c"># Pack MU count data (M and U columns)</span>
yame pack <span class="nt">-f3</span> methylation_counts.txt <span class="o">&gt;</span> yourfile.cg
</code></pre></div></div> <p><strong>2. Summarize methylation data</strong></p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>yame summary yourfile.cg
</code></pre></div></div> <p><strong>3. Test enrichment over genomic features</strong></p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>yame summary <span class="nt">-m</span> ChromHMM_states.cm yourfile.cg <span class="o">&gt;</span> enrichment_results.txt
</code></pre></div></div> <p><strong>4. Subset samples from multi-sample data</strong></p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>yame subset <span class="nt">-l</span> sample_list.txt yourfile.cg <span class="o">&gt;</span> subset.cg
</code></pre></div></div> <p><strong>5. Merge single cells into pseudobulks</strong></p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>yame subset <span class="nt">-l</span> cluster1_cells.txt single_cell.cg | yame rowop - <span class="nt">-o</span> binasum <span class="o">&gt;</span> pseudobulk.cg
</code></pre></div></div><hr /> <h2 id="cx-format-family"> <a href="#cx-format-family" class="anchor-heading" aria-labelledby="cx-format-family"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> CX Format Family </h2> <p>YAME uses the <strong>CX format family</strong> to efficiently store different types of methylation-related data:</p> <div class="table-wrapper"><table> <thead> <tr> <th>Format</th> <th>Extension</th> <th>Data Type</th> <th>Use Case</th> </tr> </thead> <tbody> <tr> <td>Format 0</td> <td><code class="language-plaintext highlighter-rouge">.cg</code></td> <td>Binary (1 byte/8 CpGs)</td> <td>Presence/absence of methylation</td> </tr> <tr> <td>Format 1</td> <td><code class="language-plaintext highlighter-rouge">.cg</code></td> <td>Value + RLE</td> <td>Compressed methylation values</td> </tr> <tr> <td>Format 2</td> <td><code class="language-plaintext highlighter-rouge">.cm</code></td> <td>State text + Index RLE</td> <td>Chromatin states, genomic features</td> </tr> <tr> <td>Format 3</td> <td><code class="language-plaintext highlighter-rouge">.cg</code></td> <td>MU RLE + Ladder byte</td> <td>M and U sequencing counts</td> </tr> <tr> <td>Format 4</td> <td><code class="language-plaintext highlighter-rouge">.cg</code></td> <td>Fraction + NA-RLE</td> <td>Methylation fractions</td> </tr> <tr> <td>Format 5</td> <td><code class="language-plaintext highlighter-rouge">.cg</code></td> <td>2-bits + NA-RLE</td> <td>Differential methylation (0,1,2)</td> </tr> <tr> <td>Format 6</td> <td><code class="language-plaintext highlighter-rouge">.cg</code></td> <td>2-bits boolean</td> <td>Set and universe indicators</td> </tr> <tr> <td>Format 7</td> <td><code class="language-plaintext highlighter-rouge">.cr</code></td> <td>Compressed BED</td> <td>CpG reference coordinates</td> </tr> </tbody> </table></div> <p>All CX formats use BGZF compression and share a consistent internal structure (<code class="language-plaintext highlighter-rouge">cdata_t</code> blocks), enabling uniform handling across different data types.</p><hr /> <h2 id="command-reference"> <a href="#command-reference" class="anchor-heading" aria-labelledby="command-reference"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Command Reference </h2> <p>The following commands provide the core functionality of YAME. Click on each command for detailed documentation.</p> <h3 id="data-storage--conversion"> <a href="#data-storage--conversion" class="anchor-heading" aria-labelledby="data-storage--conversion"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data Storage &amp; Conversion </h3> <ul> <li><a href="/YAME/docs/pack_unpack.html"><code class="language-plaintext highlighter-rouge">pack</code></a> - Pack data into CX format</li> <li><a href="/YAME/docs/pack_unpack.html"><code class="language-plaintext highlighter-rouge">unpack</code></a> - Unpack CX files to text format</li> </ul> <h3 id="data-subsetting"> <a href="#data-subsetting" class="anchor-heading" aria-labelledby="data-subsetting"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data Subsetting </h3> <ul> <li><a href="/YAME/docs/subset.html"><code class="language-plaintext highlighter-rouge">subset</code></a> - Extract specific samples from multi-sample files</li> <li><a href="/YAME/docs/subset.html"><code class="language-plaintext highlighter-rouge">rowsub</code></a> - Extract specific genomic regions or CpG sites</li> </ul> <h3 id="data-summarization--analysis"> <a href="#data-summarization--analysis" class="anchor-heading" aria-labelledby="data-summarization--analysis"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data Summarization &amp; Analysis </h3> <ul> <li><a href="/YAME/docs/summarize.html"><code class="language-plaintext highlighter-rouge">info</code></a> - Display basic parameters of CX files</li> <li><a href="/YAME/docs/summarize.html"><code class="language-plaintext highlighter-rouge">summary</code></a> - Calculate statistics and test enrichment over features</li> </ul> <h3 id="sample-operations"> <a href="#sample-operations" class="anchor-heading" aria-labelledby="sample-operations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Sample Operations </h3> <ul> <li><a href="/YAME/docs/sample.html"><code class="language-plaintext highlighter-rouge">index</code></a> - Index samples in multi-sample files</li> <li><a href="/YAME/docs/sample.html"><code class="language-plaintext highlighter-rouge">split</code></a> - Split multi-sample data into individual files</li> <li><a href="/YAME/docs/sample.html"><code class="language-plaintext highlighter-rouge">chunk</code></a> - Divide data into smaller fragments</li> <li><a href="/YAME/docs/sample.html"><code class="language-plaintext highlighter-rouge">chunkchar</code></a> - Chunk text data into fragments</li> </ul> <h3 id="row-operations"> <a href="#row-operations" class="anchor-heading" aria-labelledby="row-operations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Row Operations </h3> <ul> <li><a href="/YAME/docs/rowop.html"><code class="language-plaintext highlighter-rouge">rowop</code></a> - Perform row-wise operations (mean, sum, std, binarization, co-methylation)</li> </ul> <h3 id="data-manipulation"> <a href="#data-manipulation" class="anchor-heading" aria-labelledby="data-manipulation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data Manipulation </h3> <ul> <li><a href="/YAME/docs/dsample.html"><code class="language-plaintext highlighter-rouge">mask</code></a> - Mask methylation data by setting masked records to M=U=0</li> <li><a href="/YAME/docs/dsample.html"><code class="language-plaintext highlighter-rouge">dsample</code></a> - Downsample to a fixed number of sites</li> </ul> <h3 id="differential-analysis"> <a href="#differential-analysis" class="anchor-heading" aria-labelledby="differential-analysis"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Differential Analysis </h3> <ul> <li><a href="/YAME/docs/enrichment.html"><code class="language-plaintext highlighter-rouge">pairwise</code></a> - Built-in differential methylation calling</li> </ul><hr /> <h2 id="tutorials--workflows"> <a href="#tutorials--workflows" class="anchor-heading" aria-labelledby="tutorials--workflows"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Tutorials &amp; Workflows </h2> <ul> <li><strong><a href="/YAME/docs/enrichment.html">Enrichment Testing</a></strong> - Test methylation enrichment across genomic features</li> <li><strong><a href="/YAME/docs/pack_unpack.html">Data Storage &amp; Format Conversion</a></strong> - Working with CX formats</li> <li><strong><a href="/YAME/docs/subset.html">Subsetting Data</a></strong> - Extract samples and regions</li> <li><strong><a href="/YAME/docs/summarize.html">Summarization &amp; Analysis</a></strong> - Calculate statistics and aggregations</li> <li><strong><a href="/YAME/docs/rowop.html">Row Operations</a></strong> - Merge pseudobulks and perform calculations</li> <li><strong><a href="/YAME/docs/sample.html">Sample Management</a></strong> - Handle multi-sample datasets</li> <li><strong><a href="/YAME/docs/dsample.html">Downsampling</a></strong> - Test methods at different sparsity levels</li> </ul><hr /> <h2 id="reference"> <a href="#reference" class="anchor-heading" aria-labelledby="reference"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Reference </h2> <p><strong>Goldberg*, Fu*, Atkins, Moyer, Lee, Deng, Zhouâ€  (2025)</strong><br /> KnowYourCG: Facilitating Base-level Sparse Methylome Interpretation.<br /> <em>Science Advances</em><br /> DOI: <a href="https://www.science.org/doi/10.1126/sciadv.adw3027">10.1126/sciadv.adw3027</a></p><hr /> <h2 id="acknowledgements"> <a href="#acknowledgements" class="anchor-heading" aria-labelledby="acknowledgements"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Acknowledgements </h2> <p>This work is supported by <strong>NIH/NIGMS 5R35GM146978</strong>.</p> <p>YAME integrates with the <a href="https://github.com/zhou-lab/KYCG">KYCG knowledge base</a> for comprehensive methylation feature analysis.</p><hr /> <h2 id="getting-help"> <a href="#getting-help" class="anchor-heading" aria-labelledby="getting-help"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Getting Help </h2> <ul> <li><strong>Documentation</strong>: Browse the menu for detailed guides</li> <li><strong>Command Help</strong>: Run <code class="language-plaintext highlighter-rouge">yame</code> or <code class="language-plaintext highlighter-rouge">yame &lt;command&gt;</code> for usage information</li> <li><strong>Issues</strong>: Report bugs on <a href="https://github.com/zhou-lab/YAME/issues">GitHub Issues</a></li> <li><strong>KYCG Resources</strong>: Download reference coordinates and feature files from <a href="https://github.com/zhou-lab/KYCGKB_hg38">KYCG hg38</a> or <a href="https://github.com/zhou-lab/KYCGKB_mm10">KYCG mm10</a></li> </ul><hr /> </main> </div> </div> <div class="search-overlay"></div> </div> </body> </html>
